src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -802,7 +802,7 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
             {
                 foreach (var modelError in modelState.Errors)
                 {
-                    var errorText = ValidationHelpers.GetUserErrorMessageOrDefault(modelError, modelState: null);
+                    var errorText = ValidationHelpers.GetUserErrorMessageOrDefault(modelError, modelState);
```

Comment:
Let's start with fixing that issue.

We can research why the validation summary has historically been shorter than the sum of all validation messages and tackle that separately. Worth filing a separate Investigate item for that.